_posts/2022-XX-XX-fido2-central-auth.md

```diff
@@ -0,0 +1,130 @@
+---
+layout:     post
+title:      "Local authentication in a centralized management system with FIDO2"
+date:       2022-09-08 12:00:00 +0100
+categories: idm
+---
+
+#TODO: change date
+
+#TODO: change for fido2 helper process for sssctl
+
+# Introduction
+
+I have been working on a feature that will allow centrally managed users to log in locally to a computer with FIDO2 keys. As the feature is still under development I've prepared an environment to test it.
+
+## Dependencies
+
+The test environment uses docker/podman to set up the containers and ansible to provision them. On top of that, yubikey-manager is needed to manage the FIDO2 device options. Moreover, I've created a [COPR repository](https://copr.fedorainfracloud.org/coprs/ipedrosa/fido2-central-auth/) to deliver the patched versions of SSSD and IPA that will enable this feature. Finally, I've prepared a new [sssd-ci-container branch](https://github.com/ikerexxe/sssd-ci-containers/tree/fido2) to automate the provision of everything. The following sections will explain how to set up the environment and how to test it.
+
+# How to set up the environment
+
+1. Clone sssd-ci-containers locally:
+
+```console
+git clone https://github.com/ikerexxe/sssd-ci-containers/
+git checkout --track origin/fido2
+```
+
+2. Host package installation and configuration. The following commands should be executed only once:
+
+```console
+$ sudo dnf install -y podman podman-docker docker-compose yubikey-manager
+$ sudo systemctl enable --now podman.socket
+$ sudo setsebool -P container_manage_cgroup true
+$ cp env.example .env
+```
+
+3. Open the cloned sssd-ci-containers project:

```

Comment:
This was not needed, as I had to cd into this dir above to run `git checkout` successfully